### PR TITLE
posix: split FILE_PAGE_SIZE

### PR DESCRIPTION
**Pull request by marcinslusarz**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/167)

### CODE REVIEW

**Comment by krzycz** [comment]
:lgtm:

---

Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/167#-:-KlcP-2x078l_Bz77xff:bnfp4nl)*

**Comment by marcinslusarz** [comment]
I think this one is quite good - it doesn't say everything, but it describes the core of its use. And it's nice counterpart to MAX_BLOCK_SIZE (which we already have).

If you have a better idea, then please tell me.

**Comment by GBuella** [comment]
Well, I still like `PMEMFILE_BLOCK_ALIGNMENT`. So, perhaps we could keep `MIN_BLOCK_SIZE`, and also introduce `PMEMFILE_BLOCK_ALIGNMENT`.  I'm thinking about using it in these places:
```
ASSERT(count % MIN_BLOCK_SIZE == 0);
ASSERT(pmemfile_posix_block_size % MIN_BLOCK_SIZE == 0);
return n & ~(MIN_BLOCK_SIZE - 1);
*length += *offset % MIN_BLOCK_SIZE;
 *offset -= *offset % FILE_PAGE_SIZE;
```
Because, well, a minimum size is usually not something one has to divide by.
So, perhaps in `src/libpmemfile-posix/data.c` these two lines:
```
ASSERT(count > 0);
ASSERT(count % MIN_BLOCK_SIZE == 0);
```
could be changed to something like:
```
ASSERT(count >= MIN_BLOCK_SIZE);
ASSERT(count % PMEMFILE_BLOCK_ALIGNMENT == 0);
```
What do you think?

**Comment by marcinslusarz** [comment]
Ok. I replace FILE_PAGE_SIZE with 2 constants (which now have the same value, but they don't need to be in the future) - MIN_BLOCK_SIZE and BLOCK_ALIGNMENT.

File: src/libpmemfile-posix/internal.h [hunk 1]
```diff
@@ -49,27 +49,32 @@ pmemfile_tx_abort(int err)
...
  * part of the on-media layout.
  * But later the code might (probably will) depend on this.
  */
-#define FILE_PAGE_SIZE ((size_t)0x1000)
+#define MIN_BLOCK_SIZE ((size_t)0x1000)
 
-#define MAX_BLOCK_SIZE (UINT32_MAX - (UINT32_MAX % FILE_PAGE_SIZE))
+#define BLOCK_ALIGNMENT ((size_t)0x1000)
+
+COMPILE_ERROR_ON(MIN_BLOCK_SIZE < BLOCK_ALIGNMENT);
```
Comment (GBuella) [inline src/libpmemfile-posix/internal.h]: I guess you meant `COMPILE_ERROR_ON(MIN_BLOCK_SIZE <= BLOCK_ALIGNMENT);`

File: src/libpmemfile-posix/data.c [hunk 2]
```diff
@@ -197,13 +197,13 @@ file_allocate_block_data(PMEMfilepool *pfp,
 {
 	ASSERT_IN_TX();
 	ASSERT(count > 0);
```
Comment (GBuella) [inline src/libpmemfile-posix/data.c]: I think this line should also be changed to:
`ASSERT(count >= MIN_BLOCK_SIZE)`

File: src/libpmemfile-posix/internal.h (hunk 1 above)
Comment (GBuella) [inline src/libpmemfile-posix/internal.h]: I think this is not even needed, the second check is enough.
Perhaps a check for MIN_BLOCK_SIZE being larger than zero.

Comment (marcinslusarz) [inline src/libpmemfile-posix/internal.h]: With <= it would trigger right now.

Comment (GBuella) [inline src/libpmemfile-posix/internal.h]: oh, right, me silly.

File: src/libpmemfile-posix/data.c (hunk 2 above)
Comment (marcinslusarz) [inline src/libpmemfile-posix/data.c]: Sure.

File: src/libpmemfile-posix/internal.h (hunk 1 above)
Comment (GBuella) [inline src/libpmemfile-posix/internal.h]: Well than, my second comment still stands.

Comment (marcinslusarz) [inline src/libpmemfile-posix/internal.h]: Indeed.

Comment (marcinslusarz) [inline src/libpmemfile-posix/internal.h]: Done. I can't imagine anyone proposing (or even testing) MIN_BLOCK_SIZE to become 0.

**Comment by GBuella** [comment]
:lgtm:

---

Reviewed 4 of 6 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/167#-:-KldI1kjC9DXYRbcKoi9:bnfp4nl)*

**Pull request merged**